earthmover/init.py

```diff
@@ -0,0 +1,43 @@
+import os
+from pathlib import Path
+import shutil
+
+def populate_project(project_name: str) -> None:
+    root_path = Path(os.path.dirname(__file__))
+    include_path = root_path / "include" / "starter_project"
+
+    # initialize starter project
+    project_path = Path(project_name)
+    project_path.mkdir()
+    shutil.copy2(include_path / "README.md", project_path)
+    shutil.copy2(include_path / "earthmover.yaml", project_path)
+
+    project_sources = project_path / "sources"
+    project_templates = project_path / "templates"
+    project_sources.mkdir()
+    project_templates.mkdir()
+
+    # copy example files used by test suite into starter project
+    test_path = root_path / "tests"
+    shutil.copy2(test_path / "sources" / "mammals.csv", project_sources)
+    shutil.copy2(test_path / "sources" / "fishes.csv", project_sources)
+    shutil.copy2(test_path / "templates" / "animal.jsont", project_templates)
+
+def run_init() -> None:
+    project_name = input("Enter a name for your project: ")
+
+    # sanitize
+    clean_name = "".join(c for c in project_name if (c.isalnum() or c in "_-"))
+    if len(clean_name) == 0:
+        print(f"ERROR: entered name has no valid characters (alphanumeric, underscore, dash)")
+        return None
+
+    full_path = Path(os.getcwd(), clean_name)
+
+    try:
+        populate_project(full_path)
+    except FileExistsError:
```

Comment:
@johncmerfeld or @tomreitz Instead of returning `None` here and checking for a none to know to `exit(1)` in main, why not just raise the exit here?

Reply:
@jayckaiser my philosophy here is that it's easier to reason about an application if unhandled exceptions / exits are only thrown from one place, usually the top-level function. This also makes `run_init` simpler and more portable.

We don't reap many tangible benefits from such a design in this case but I think it's a good principle.

Reply:
This principle makes sense and I agree to its adherence.